### PR TITLE
util/signature: add WalletConnect API support

### PR DESCRIPTION
**Pull request by fyrchik**
Signed-off-by: Evgenii Stratonikov <evgeniy@nspcc.ru>

### CODE REVIEW

File: util/signature/walletconnect/sign_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+func TestSignMessage(t *testing.T) {
+	p1, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	require.NoError(t, err)
+
+	msg := []byte("NEO")
+	result, err := SignMessage(p1, msg)
+	require.NoError(t, err)
+	//require.Equal(t, elliptic.MarshalCompressed(elliptic.P256(), p1.PublicKey.X, p1.PublicKey.Y), result.PublicKey)
```
Comment (carpawell) [inline util/signature/walletconnect/sign_test.go:20]: debug?

File: util/signature/walletconnect/sign.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+func createMessageWithSalt(msg, salt []byte) []byte {
+	// 4 byte prefix + length of the message with salt in bytes +
+	// + salt + message + 2 byte postfix.
+	saltedLen := len(salt)*2 + len(msg)
+	data := make([]byte, 4+getVarIntSize(saltedLen)+saltedLen+2)
+
+	n := copy(data, []byte{0x01, 0x00, 0x01, 0xf0})         // fixed prefix
+	n += putVarUint(data[n:], uint64(len(salt)*2+len(msg))) // salt is hex encoded, double its size
```
Comment (alexvanin) [inline util/signature/walletconnect/sign.go:102]: I guess we can use `saltedLen` there. Overlooked it at first.

Comment (fyrchik) [reply util/signature/walletconnect/sign.go]: Fixed

**Review by alexvanin — DISMISSED** [review]
go.mod changes look a bit strange, though

**Comment by carpawell** [comment]
A binary in the PR. Was it planned?

File: util/signature/walletconnect/sign.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+func Sign(p *ecdsa.PrivateKey, msg []byte) ([]byte, error) {
+	sm, err := SignMessage(p, msg)
+	if err != nil {
+		return nil, err
+	}
+	return append(sm.Data, sm.Salt...), nil
+}
+
+// Verify verifies message using WalletConnect API. The returned signature
+// contains RFC6979 signature and 16-byte salt.
```
Comment (carpawell) [inline util/signature/walletconnect/sign.go:40]:
```suggestion
// Verify verifies message using WalletConnect API.
```

copypaste?

**Review by carpawell — CHANGES_REQUESTED** [review]

File: util/signature/walletconnect/sign.go (hunk 3 above)
Comment (fyrchik) [reply util/signature/walletconnect/sign.go:40]: It is nice to have some info about the signature format.

Comment (carpawell) [reply util/signature/walletconnect/sign.go:40]: yes, that is good but what about "...returned..."? func does not return any signature

Comment (fyrchik) [reply util/signature/walletconnect/sign.go:40]: Fixed

**Review by carpawell — APPROVED** [review]

**Pull request merged**